frontend/src/components/misc/SearchBar.tsx

```diff
@@ -0,0 +1,106 @@
+import { Checkbox, Input } from 'antd';
+import { AnimatePresence } from 'framer-motion';
+import { observer } from 'mobx-react';
+import React, { Component } from 'react'
+import { uiSettings } from '../../state/ui';
+import { animProps_span_searchResult, MotionSpan } from '../../utils/animationProps';
+import { FilterableDataSource } from '../../utils/filterableDataSource';
+
+// todo: extract out where the filterText is retreived from / saved.
+//       this component was originally extracted out of another component, but we probably want to re-use it elsewhere in the future
+@observer
+class SearchBar<TItem> extends Component<{ dataSource: () => TItem[], isFilterMatch: (filter: string, item: TItem) => boolean }> {
+
+    private filteredSource = {} as FilterableDataSource<TItem>;
+    get data() { return this.filteredSource.data; }
+
+    /*
+        todo: autocomplete:
+        - save as suggestion on focus lost, enter, or clear
+        - only show entries with matching start
+    */
+    // todo: allow setting custom "rows" to search, and case sensitive or not (pass those along to isFilterMatch)
+
+    constructor(p: any) {
+        super(p);
+        this.filteredSource = new FilterableDataSource<TItem>(this.props.dataSource, this.props.isFilterMatch);
+        this.filteredSource.filterText = uiSettings.topicList.quickSearch;
+    }
+
+    componentWillUnmount() {
+        this.filteredSource.dispose();
+    }
+
+    render() {
+        return <div style={{ marginBottom: '.5rem', padding: '0', whiteSpace: 'nowrap', display: 'flex', alignItems: 'center' }}>
+            {/* <AutoComplete placeholder='Quick Search' size='large'
+                style={{ width: 'auto', padding: '0' }}
+                onChange={v => this.filteredSource.filterText = String(v)}
+                dataSource={['battle-logs', 'customer', 'asdfg', 'kafka', 'some word']}
+            > */}
+            <Input allowClear={true} placeholder='Quick Search' size='large' style={{ width: '350px' }}
+                onChange={e => this.filteredSource.filterText = uiSettings.topicList.quickSearch = e.target.value}
```

Comment:
Careful with this. This is a remnant that needs to be cleaned up first.
Every instance of a SearchBar component should obviously have its own place where it saves its current value 😅  
Currently the variable is hardcoded (uiSettings.topicList.quickSearch).

The most obvious idiomatic solution would be to make the SearchBar "controlled" (in React lingo). Meaning that it would expose `value` and `onChange` (just like `<Input .../>` does), and then the caller (TopicList etc) would wire those two things up to the correct variable.

Reply:
Yes, you're obviously right. I missed that one while extracting. Gonna go fix it.

Reply:
Please have another look, I moved on with your suggestion. I think I did not miss anything this time.